bolt-sidecar/src/primitives/commitment.rs

```diff
@@ -173,10 +176,31 @@ impl InclusionRequest {
     /// Validates the priority fee against a minimum priority fee.
     /// Returns `true` if the "effective priority fee" is greater than or equal to the set minimum
     /// priority fee, `false` otherwise.
-    pub fn validate_min_priority_fee(&self, max_base_fee: u128, min_priority_fee: u128) -> bool {
-        self.txs.iter().all(|tx| {
-            tx.effective_tip_per_gas(max_base_fee).map_or(false, |tip| tip >= min_priority_fee)
-        })
+    /// Returns an error if min priority fee cannot be calculated.
+    pub fn validate_min_priority_fee(
```

Comment:
I would move the bulk logic of this function to `ExecutionState`, and just have this validate that the `effective_tip_per_gas` is greater than the result of running the pricing algo. So the function can actually remain the same as before.